### PR TITLE
Document classes that moved

### PR DESCRIPTION
**Pull request by rgstephens**
**Proposed changes**:
- Add docs on classes that moved to the Migration page.

**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [x] updated the documentation
- [ ] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/master/changelog) for instructions)
- [ ] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

**Comment by akelad** [comment]
requested Tobi's review because he prob has a better overview of these changes

File: docs/docs/migration-guide.mdx [hunk 1]
```diff
@@ -594,9 +593,23 @@ policies:
...
+| Rasa 1.0                             | Rasa 2.0                          |
+| ------------------------------------ | --------------------------------- |
+| rasa.core.events                     | rasa.shared.core.events           |
+| rasa.core.trackers                   | rasa.shared.core.trackers         |
+| rasa.core.channels.channel.RestInput | rasa.core.channels.rest.RestInput |
+| Rasa 1.0                             | Rasa 2.0                          |
+| ------------------------------------ | --------------------------------- |
+| rasa.core.events                     | rasa.shared.core.events           |
+| rasa.core.trackers                   | rasa.shared.core.trackers         |
+| rasa.core.channels.channel.RestInput | rasa.core.channels.rest.RestInput |
```
Comment (wochinge) [inline docs/docs/migration-guide.mdx:611]: duplicate
```suggestion
```

Comment (wochinge) [inline docs/docs/migration-guide.mdx:611]: this one is technically not related to `rasa.shared`

File: docs/docs/migration-guide.mdx [hunk 2]
```diff
@@ -594,9 +593,23 @@ policies:
...
     featurizer:
     - name: SingleStateFeaturizer
-
 ```
 
+### Class changes
+
+The following classes have moved from `rasa.core` to `rasa.shared.core`:
+
+| Rasa 1.0                             | Rasa 2.0                          |
```
Comment (wochinge) [inline docs/docs/migration-guide.mdx:602]: Other notable changes due to `rasa.shared`: 
* `rasa.core.Domain`
* `rasa.core.importers`
* `rasa.nlu.training_data`

**Review by wochinge — COMMENTED** [review]
Thanks for the PR! I made a couple of suggestions about further things to include. I'm curious: How did this come up?

File: docs/docs/migration-guide.mdx (hunk 1 above)
Comment (rgstephens) [reply docs/docs/migration-guide.mdx:611]: This request came from a customer who was upgrading their bot to 2.0 and had to change their code.

Comment (rgstephens) [reply docs/docs/migration-guide.mdx:611]: I just asked in [slack](https://rasacustomers.slack.com/archives/CSCRVPBBK/p1604080748017700)

**Comment by wochinge** [comment]
@rgstephens Do you have time to wrap this up?

**Comment by wochinge** [comment]
@rgstephens can we please wrap this up?
